classes/class-ledyer-main.php

```diff
@@ -158,6 +158,11 @@ public function update_order_status( $request ) {
 						if( is_wp_error( $response ) ) {
 							\Ledyer\Logger::log( 'Couldn\'t acknowledge order ' . $order_id  );
 						}
+						
+						$ledyer_update_order = ledyer()->api->update_order_reference( $order_id, array( 'reference' => strval( $order->ID ) ) );
+						if ( is_wp_error( $ledyer_update_order ) ) {
+							\Ledyer\Logger::log( 'Couldn\'t set merchant reference ' . $order->ID  );
+						}
```

Comment:
Moved setting order reference to inside where the receive the notification from Leder BE, instead of when the confirmation page is triggered by FE. It makes more sense to store such logic here, where we also do the ack.